### DISCUSSION TITLE
cronicle not running after a reboot

### DISCUSSION THREAD

**Discussion by Drallas** [jhuckaby/Cronicle#653] [General]
Noticed that after a reboot cronicle not running:

```
/opt/cronicle/bin/control.sh status
status: Cronicle Daemon not running (no pid file)
```

```bash
cd /opt/cronicle
npm run boot

systemctl status cronicle
× cronicle.service - Node Cronicle
     Loaded: loaded (/etc/systemd/system/cronicle.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Thu 2023-10-12 11:26:54 UTC; 1min 9s ago
    Process: 356 ExecStart=/opt/cronicle/bin/control.sh start (code=exited, status=3)
        CPU: 3ms

Oct 12 11:26:54 backupserver systemd[1]: Starting cronicle.service - Node Cronicle...
Oct 12 11:26:54 backupserver control.sh[356]: /opt/cronicle/bin/control.sh start: Starting up Cronicle Daemon...
Oct 12 11:26:54 backupserver control.sh[356]: /opt/cronicle/bin/control.sh: 67: node: not found
Oct 12 11:26:54 backupserver control.sh[356]: /opt/cronicle/bin/control.sh start: Cronicle Daemon could not be started
Oct 12 11:26:54 backupserver systemd[1]: cronicle.service: Control process exited, code=exited, status=3/NOTIMPLEMENTED
Oct 12 11:26:54 backupserver systemd[1]: cronicle.service: Failed with result 'exit-code'.
Oct 12 11:26:54 backupserver systemd[1]: Failed to start cronicle.service - Node Cronicle.
```

`Linux backupserver 6.1.0-13-cloud-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux`

**Comment by jhuckaby** [comment]
Hmmm, so here is the problem:

```
Oct 12 11:26:54 backupserver control.sh[356]: /opt/cronicle/bin/control.sh: 67: node: not found
```

Somehow your Node.js binary isn't in the `PATH` environment variable when your server is starting up.  That's very odd.  I've never seen this issue before.

Did you install Node.js somewhere custom on your server?  Try finding out where your Node.js lives (usually in `/usr/bin` or `/usr/local/bin`).

I'll work on trying to fix this from my end, perhaps cooking up a default PATH environment variable with some common locations in the control script.

**Reply by Drallas** [reply]
It's here echo $PATH
/usr/local/lib/nodejs/node-v18.18.1-linux-x64/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games

followed https://github.com/nodejs/help/wiki/Installation

where do you expect node.js?

**Reply by jhuckaby** [reply]
Okay, so that is a custom location.  Your Linux systemd cannot find it there on boot.  It should really be in `/usr/bin/` ideally.

**Reply by Drallas** [reply]
Thanks for now, I created a symbolic link `/usr/bin/node`

```bash
/usr/bin/node --version
v18.18.1
```

Service is running.

```bash
systemctl status cronicle
● cronicle.service - Node Cronicle
     Loaded: loaded (/etc/systemd/system/cronicle.service; enabled; preset: enabled)
     Active: active (running) since Thu 2023-10-12 21:36:07 CEST; 1s ago
    Process: 33373 ExecStart=/opt/cronicle/bin/control.sh start (code=exited, status=0/SUCCESS)
   Main PID: 33384 (Cronicle Server)
      Tasks: 11 (limit: 4691)
     Memory: 37.7M
        CPU: 514ms
     CGroup: /system.slice/cronicle.service
             └─33384 "Cronicle Server"

Oct 12 21:36:07 backupserver systemd[1]: Starting cronicle.service - Node Cronicle...
Oct 12 21:36:07 backupserver control.sh[33373]: /opt/cronicle/bin/control.sh start: Starting up Cronicle Daemon...
Oct 12 21:36:07 backupserver control.sh[33373]: /opt/cronicle/bin/control.sh start: Cronicle Daemon started
Oct 12 21:36:07 backupserver systemd[1]: Started cronicle.service - Node Cronicle.
```

Thanks for your help.

**Comment by jhuckaby** [comment]
I just released [v0.9.36](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.36) which should help with this issue, assuming your Node.js is installed to a standard location (e.g. `/usr/bin/`).

https://github.com/jhuckaby/Cronicle/commit/ca1ae437ec4fc9af472e24bd076346ca17392283

**Reply by Drallas** [reply]
Great:
`sudo /opt/cronicle/bin/control.sh upgrade`
On the latest now!

**Comment by sirjmann92** [comment]
This is apparently still an issue in v0.9.4 - I had to do the symlink thing as well.

I've hunted off and on for months and tried many things to get Cronicle to start with server boot, nothing worked until I found this thread. It's been so long now since I first installed it, I can't even remember which instructions I followed to install node.js but it obviously didn't add itself to my $PATH. At least it's not in $PATH when Cronicle tries to start. The symlink workaround works... 

Followed this for creating the symlink: https://gist.github.com/MeLlamoPablo/0abcc150c10911047fd9e5041b105c34 - Cronicle started successfully on the next reboot after this.

Node was installed here: /home/MYUSER/.nvm/versions/node/v20.10.0/bin
and here: /root/.nvm/versions/node/v20.10.0/bin

I'm a Linux noob so I probably missed something in some instructions somewhere... just posting to let @jhuckaby know, in case there's something that can be done for other users that find themselves in a similar situation.

**Reply by jhuckaby** [reply]
I'll add a note in the docs about this.  Thank you!

**Reply by feerlessleadr** [reply]
Just wanted to say thanks for this. I had 2 separate fresh installs of node & chronicle on 2 separate ubuntu machines have this issie, and this comment fixed it for me.
